os error 22 while copying a simple directory

I encountered the following error when copying a directory that contains only a text file.
I'm using CentOS 7.5. `rmz` works just fine.

```
user@hostname:~$ cpz --version
cpz 1.1.10
user@hostname:~$ rustc --version
rustc 1.75.0 (82e1608df 2023-12-21)
user@hostname:~$ cpz test test2
Error: An I/O error occurred
│
╰─▶ Invalid argument (os error 22)
    ╰╴Failed to copy file: "test/abc.txt"
user@hostname:~$ cat test/abc.txt
ABC
user@hostname:~$ ls test
abc.txt
user@hostname:~$ 
```

My uname -a outputs `Linux hostname 4.14.78-9.x86_64 #18 SMP Tue Feb 2 07:46:57 PST 2021 x86_64 x86_64 x86_64 GNU/Linux`, it's a 64 bit system.

I encountered the following error while trying to run "cargo b" under cpz directory:

```
   Compiling fuc_engine v1.1.6 (/home/scratch.shachang_vlsi_2/github/fuc/fuc_engine)                                         
error[E0308]: mismatched types
   --> fuc_engine/src/ops/copy.rs:401:63      
    |                                                                                                                  
401 |                 match copy_file_range(&from, None, &to, None, isize::MAX - total_copied) {                             
    |                       ---------------                         ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `isize`                                                                                                                             
    |                       |           
    |                       arguments to this function are incorrect                                                         
    |                                         
note: function defined here                                                                                            
   --> /home/shachang/.nvidia-utils/x86_64-redhat-linux/cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.28/src/fs/copy_file_range.rs:12:8                                                                                                 
    |                                                         
12  | pub fn copy_file_range<InFd: AsFd, OutFd: AsFd>(        
    |        ^^^^^^^^^^^^^^^                                  
help: you can convert an `isize` to a `usize` and panic if the converted value doesn't fit                                   
    |                                                         
401 |                 match copy_file_range(&from, None, &to, None, (isize::MAX - total_copied).try_into().unwrap()) {       
    |                                                               +                         +++++++++++++++++++++          

error[E0308]: mismatched types                                
   --> fuc_engine/src/ops/copy.rs:416:29                      
    |                                                         
416 |             total_copied += byte_copied;                
    |                             ^^^^^^^^^^^ expected `isize`, found `usize`                                                

error[E0277]: cannot add-assign `usize` to `isize`            
   --> fuc_engine/src/ops/copy.rs:416:26                      
    |                                                         
416 |             total_copied += byte_copied;                
    |                          ^^ no implementation for `isize += usize`                                                     
    |                                                         
    = help: the trait `AddAssign<usize>` is not implemented for `isize`                                                      
    = help: the following other types implement trait `AddAssign<Rhs>`:                                                      
              <isize as AddAssign>                            
              <isize as AddAssign<&isize>>                    
```

Then, I changed `isize` to `usize` like what you suggested. It can build without problem. But still it has problem when running the same experiment:

```
user@hostname:~/scratch/github/fuc/target/debug$ ./cpz test test2
Error: An I/O error occurred
├╴at cpz/src/main.rs:70:18
│
╰─▶ Invalid argument (os error 22)
    ├╴at cpz/src/main.rs:68:45
    ╰╴Failed to copy file: "test/abc.txt"
```

--- a/fuc_engine/src/ops/copy.rs
+++ b/fuc_engine/src/ops/copy.rs
@@ -398,7 +398,7 @@ mod compat {
         let mut total_copied = 0;
         loop {
             let byte_copied =
-                match copy_file_range(&from, None, &to, None, usize::MAX - total_copied) {
+                match copy_file_range(&from, None, &to, None, usize::MAX / 2 - total_copied) {
                     Err(Errno::XDEV) if total_copied == 0 => {
                         return copy_any_file(from, to, file_name, from_path);
                     }

